### PR TITLE
Make the health transfer from #25326 into a weighted average to enable balance tweaking

### PR DESCRIPTION
**Pull request by PolterTzi**
## About the PR

Changes the transfer of health from space-wizards/space-station-14/pull/25326
Instead of taking the new health of the clipped plant, it instead does a weighted average of:
Its health before clipping
Its health after clipping
The Endurance of the plant (aka base health)

## Why / Balance

This would make the clipping somewhat more viable in case the last change was too harsh on it. Since the average is weighted, it should be relatively easy to tune the values.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**

:cl:
- tweak: Tweaked the health of seeds from clipped plants to be somewhat less punishing.

### CODE REVIEW

**Comment by PolterTzi** [comment]
Changed to draft, since the weights are nowhere near final.

**Comment by ViceEmargo** [comment]
After finally having a chance to try the clipper change from 25326, I noticed you could not continue to keep a plant's "lineage" alive without dumping a ton of nutriment or another chemical that heals the plant.
You start with a plant at 100 health, it gets clipped, then you have 2 plants at ~60 health. By the time the new plant grows, you'll have a 80 health plant, clipping that again to continue its lineage, you'll have 2 ~40 health plants. When this plant is ready to harvest, it will have about 60 health. At this point it will be impossible to keep its lineage going without dumping a chemical which raises the health.
I'm hoping this change will alleviate this.
For clarification, with this PR, does clipping a plant reduce the host plant's health?

**Comment by PolterTzi** [comment]
> For clarification, with this PR, does clipping a plant reduce the host plant's health?

The health of both plants are still reduced. This aims to reduce the damage done to the newly created seed by making it a weighted average of a few factors, rather than directly copying the host's damaged health. The weights in the commit itself are placeholders, they're meant to be adjusted to reflect desired balance better.

**Comment by Partmedia** [comment]
I'm not sure how I feel about this, because:

1. Why add more complexity (like this weighted average) when instead you can tweak the health change for clipping?
2. I'm not convinced that the health change needs to be reduced. The seedless trait is intended to make it more difficult to create a "lineage" of an extremely potent plant. Can you explain why it needs to be easier? If you can convince me, the easier way to deal with this would just be to reduce the probability of getting the seedless trait.

**Comment by PolterTzi** [comment]
> Why add more complexity (like this weighted average) when instead you can tweak the health change for clipping?

Fair point. Partly I guess felt like a better idea to adjust the new mechanic than the old stuff and this seemed like a natural and "tweakable" way to reel back on the added damage without changing the "base" effect.

> I'm not convinced that the health change needs to be reduced.

Admittedly, I am not good at discerning balance or arguing too well about it, so I don't think I could personally do much convincing here.

**Comment by Partmedia** [comment]
> Fair point. Partly I guess felt like a better idea to adjust the new mechanic than the old stuff and this seemed like a natural and "tweakable" way to reel back on the added damage without changing the "base" effect.

I think that if you wanted to tweak it, the health change is the number to tweak, rather than adding another layer of mechanics on top of it.

> Admittedly, I am not good at discerning balance or arguing too well about it, so I don't think I could personally do much convincing here.

Let me rephrase it this way: *why* do you (or others) want to make this change that you're proposing? Would those changes still be necessary if something else was tweaked instead (e.g. seedless became less probable)? Then, you (or others) need to show, with evidence and/or justification, why it would make the game more fun.

**Comment by K-Dynamic** [comment]
> Let me rephrase it this way: _why_ do you (or others) want to make this change that you're proposing? Would those changes still be necessary if something else was tweaked instead (e.g. seedless became less probable)? Then, you (or others) need to show, with evidence and/or justification, why it would make the game more fun.

There's concern over successive plants with permanently lower max health each generation without loads of cryox or nutrimeat, which seems unnatural - seedlings from generations of excessively preemptive grafts should start damaged but not be capped as it matures

**Comment by Partmedia** [comment]
I'm opting to close this, because it would be better to first investigate tweaking the actual balance knob, namely the random health change from sampling. If after you determine that there is no way to balance that, feel free to re-open discussion on this.
